subsys/net/lib/lwm2m/lwm2m_obj_firmware_pull.c

```diff
@@ -464,3 +469,10 @@ int lwm2m_firmware_start_transfer(char *package_uri)
 
 	return 0;
 }
+
+#if defined(CONFIG_LWM2M_FIRMWARE_UPDATE_PULL_COAP_PROXY_SUPPORT)
+void lwm2m_firmware_set_proxy(char *in_cfg_proxy_uri)
+{
+	cfg_proxy_uri = in_cfg_proxy_uri;
```

Comment:
We probably would need to add the NULL check here and if so re-assign cfg_proxy_uri to CONFIG_LWM2M_FIRMWARE_UPDATE_PULL_COAP_PROXY_ADDR.